tw-tkms-starter/src/test/java/com/transferwise/kafka/tkms/test/BaseIntTest.java

```diff
@@ -33,12 +33,16 @@ public void cleanup() {
     tkmsSentMessagesCollector.clear();
 
     for (Meter meter : meterRegistry.getMeters()) {
-      if (!(meter instanceof Gauge)) {
+      if (!(meter instanceof Gauge) && !(isKafkaProducerMeter(meter))) {
```

Comment:
I guess this is related to error you were having running the test isolated vs running all of them. Why did the tests remove everything but gauges?

Reply:
The tests in the repo needs the cleanup, it seems that those tests were built with this assumption, not sure why. I excluded the removing the Kafka Producer metrics because the same producer is used in many tests, so if we remove it before the new test checked for the metrics the test will fail. Also there's no need to remove the Kafka Producer metrics.